### PR TITLE
Refactor advanced security scanners

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/audit/jas/applicabilitymanager_test.go [hunk 1]
```diff
@@ -349,19 +432,12 @@ func TestParseResults_AllCvesNotApplicable(t *testing.T) {
...
-	analyzerManagerExecuter = &analyzerManagerMock{}
-
 	// Act
+	assert.NoError(t, rtutils.DownloadAnalyzerManagerIfNeeded())
 	extendedResults, err := GetExtendedScanResults(fakeBasicXrayResults, fakeBasicDependencyGraph, &fakeServerDetails, []coreutils.Technology{coreutils.Npm}, nil)
 
 	// Assert
 	assert.Error(t, err)
-	assert.Equal(t, "failed to run Applicability scan. Exit code received: analyzer manager failure message", err.Error())
+	assert.Contains(t, err.Error(), "failed to run Applicability scan")
```
Comment (sverdlov93) [inline xray/audit/jas/applicabilitymanager_test.go:441]: assert.equalerror

File: xray/audit/jas/jasmanager.go [hunk 2]
```diff
@@ -12,50 +13,93 @@ import (
...
-	}
-	if !analyzerManagerExist {
-		log.Debug("Since the 'Analyzer Manager' doesn't exist locally, its execution is skipped.")
-		return &utils.ExtendedScanResults{XrayResults: xrayResults}, nil
+		return nil, err
 	}
-	applicabilityScanResults, eligibleForApplicabilityScan, err := getApplicabilityScanResults(xrayResults,
-		dependencyTrees, serverDetails, scannedTechnologies, workingDirs, analyzerManagerExecuter)
+	defer func() {
+		cleanup := scanner.scannerDirCleanupFunc
```
Comment (sverdlov93) [inline xray/audit/jas/jasmanager.go:80]: can be null? 
scanner.scannerDirCleanupFunc

File: xray/audit/jas/jasmanager.go [hunk 3]
```diff
@@ -12,50 +13,93 @@ import (
...
+	scanner.resultsFileName = filepath.Join(tempDir, "results.sarif")
+	scanner.workingDirs, err = utils.GetFullPathsWorkingDirs(workingDirs)
+	return
+}
+
+func (a *AdvancedSecurityScanner) Run(scannerCmd ScannerCmd) (err error) {
+	for _, workingDir := range a.workingDirs {
+		func() {
+			defer func() {
+				err = errors.Join(err, deleteJasProcessFiles(a.configFileName, a.resultsFileName))
```
Comment (sverdlov93) [inline xray/audit/jas/jasmanager.go:59]: why not a.deleteJasProcessFiles() ?

File: xray/audit/jas/jasmanager_test.go [hunk 4]
```diff
@@ -75,18 +61,17 @@ var fakeServerDetails = config.ServerDetails{
 }
 
 func TestGetExtendedScanResults_AnalyzerManagerDoesntExist(t *testing.T) {
-	// Arrange
-	analyzerManagerExists = false
-	analyzerManagerExecuter = &analyzerManagerMock{}
-
-	// Act
+	tmpDir, err := fileutils.CreateTempDir()
```
Comment (sverdlov93) [inline xray/audit/jas/jasmanager_test.go:64]: delete tempdir?

File: xray/audit/jas/secretsscanner.go [hunk 5]
```diff
@@ -32,74 +21,39 @@ type SecretScanManager struct {
...
-		return nil, false, utils.ParseAnalyzerManagerError(utils.Secrets, err)
+	if err = secretScanManager.scanner.Run(secretScanManager); err != nil {
+		err = utils.ParseAnalyzerManagerError(utils.Secrets, err)
+		return
 	}
 	if len(secretScanManager.secretsScannerResults) > 0 {
 		log.Info(len(secretScanManager.secretsScannerResults), "secrets were found")
 	}
-	return secretScanManager.secretsScannerResults, true, nil
+	results = secretScanManager.secretsScannerResults
```
Comment (sverdlov93) [inline xray/audit/jas/secretsscanner.go:35]:
```suggestion
    results = secretScanManager.secretsScannerResults
	if len(results) > 0 {
		log.Info(len(results), "secrets were found")
	}
	
```

File: xray/audit/jas/secretsscanner.go [hunk 6]
```diff
@@ -119,17 +73,17 @@ func (s *SecretScanManager) createConfigFile(currentWd string) error {
...
-	return createScannersConfigFile(s.configFileName, configFileContent)
+	return createScannersConfigFile(s.scanner.configFileName, configFileContent)
 }
 
 func (s *SecretScanManager) runAnalyzerManager() error {
-	return s.analyzerManager.Exec(s.configFileName, secretsScanCommand, s.serverDetails)
+	return s.scanner.analyzerManager.Exec(s.scanner.configFileName, secretsScanCommand, s.scanner.serverDetails)
 }
 
 func hideSecret(secret string) string {
```
Comment (sverdlov93) [inline xray/audit/jas/secretsscanner.go:89]: if len(secret) <= 3 {
		return "***"
	}
	return secret[:3] + strings.Repeat("*", len(secret)-3)

File: xray/audit/jas/secretsscanner_test.go [hunk 7]
```diff
@@ -10,106 +10,133 @@ import (
...
-	analyzerManagerExecutionError = errors.New(analyzerManagerErrorMessage)
-
-	// Act
-	secretsResults, entitledForSecrets, err := getSecretsScanResults(&fakeServerDetails, nil, &analyzerManagerMock{})
+	secretsResults, err := getSecretsScanResults(scanner)
 
 	// Assert
 	assert.Error(t, err)
-	assert.Equal(t, "failed to run Secrets scan. Exit code received: analyzer manager failure message", err.Error())
+	assert.Contains(t, err.Error(), "failed to run Secrets scan")
```
Comment (sverdlov93) [inline xray/audit/jas/secretsscanner_test.go:138]: assert.equalserrors

File: xray/utils/analyzermanager.go [hunk 8]
```diff
@@ -153,12 +128,16 @@ func GetAnalyzerManagerDirAbsolutePath() (string, error) {
...
 
-func getAnalyzerManagerExecutable() (string, error) {
+func GetAnalyzerManagerExecutable() (string, error) {
 	analyzerManagerDir, err := GetAnalyzerManagerDirAbsolutePath()
 	if err != nil {
 		return "", err
 	}
-	return filepath.Join(analyzerManagerDir, GetAnalyzerManagerExecutableName()), nil
+	analyzerManagerPath := filepath.Join(analyzerManagerDir, GetAnalyzerManagerExecutableName())
+	if exists, err := fileutils.IsFileExists(analyzerManagerPath, false); err != nil || !exists {
```
Comment (sverdlov93) [inline xray/utils/analyzermanager.go:137]: error swollowed

File: xray/audit/jas/jasmanager.go (hunk 2 above)
Comment (omerzi) [reply xray/audit/jas/jasmanager.go:80]: nope

File: xray/audit/jas/jasmanager.go (hunk 3 above)
Comment (omerzi) [reply xray/audit/jas/jasmanager.go:59]: @sverdlov93, there is no need for any of the struct parameters, so I don't think adding this function as a method will make it clearer/better.

**Pull request merged**